### PR TITLE
feat: special case purging of users previously migrated to Spanner

### PR DESCRIPTION
**Pull request by pjenvey**
don't issue deletes to their current data on syncstorage if their old "migration records" point to it

Closes SYNC-4225

### CODE REVIEW

**Comment by jrconlin** [comment]
Looks like a `cargo update` might help with the audit.

File: tools/tokenserver/purge_old_records.py [hunk 1]
```diff
@@ -108,34 +108,38 @@ def purge_old_records(
...
+                                    override_node=override_node
+                                )
+                            except requests.HTTPError:
+                                logger.warn(
+                                    "Delete failed for user "
+                                    f"{row.uid} [{row.node}]"
+                                )
+                                if override_node:
+                                    # Assume override_node should be reachable
+                                    raise
```
Comment (pjenvey) [inline tools/tokenserver/purge_old_records.py:142]: Note @jrconlin I just newly added this (and I forgot to commit it separately)

File: tools/tokenserver/purge_old_records.py [hunk 2]
```diff
@@ -183,6 +188,23 @@ def delete_service_data(user, secret, timeout=60, dryrun=False):
...
+    node) would delete their active data, which should be avoided
+    """
+    if override_node and replaced_at_row.node != override_node:
+        # NOTE: Users who never connected after being migrated could be
+        # assigned a spanner node record by get_user (TODO: rename get_user ->
+        # get_or_assign_user)
+        user = database.get_user(replaced_at_row.email)
+        if (user["generation"] == replaced_at_row.generation and
+            user["client_state"] == replaced_at_row.client_state):
+            return True
```
Comment (jrconlin) [inline tools/tokenserver/purge_old_records.py:205]:
```suggestion
        return (user["generation"] == replaced_at_row.generation and
            user["client_state"] == replaced_at_row.client_state)
```

**Comment by jrconlin** [comment]
Yeah, I kinda wonder if we should standardize to black for python stuff, since autopush wants to run both. It'd save us a bit of hassle to just have the various IDEs run black on save and be done with that sort of thing.

**Review by jrconlin — APPROVED** [review]

**Review by taddes — APPROVED** [review]
LGTM, just a few small suggestions if you want to consider.

**Comment by taddes** [comment]
How do you feel if I were to draft a quick PR to give us those same Makefile commands to run linting and style enforcement? At the least, flake8. That way it's consistent as @jrconlin  mentioned.

**Comment by jrconlin** [comment]
We're going to land this PR as is, mostly because of time constraints, but @taddes has a few good suggestions for clean up. Let's move those to a different PR (feel free to create one @taddes) and we can land that later.

**Pull request merged**